block-producer/src/state_view/mod.rs

```diff
@@ -162,10 +172,19 @@ fn ensure_in_flight_constraints(
     Ok(())
 }
 
+#[instrument(skip_all, err(Debug), fields(COMPONENT = crate::COMPONENT))]
 fn ensure_tx_inputs_constraints(
     candidate_tx: SharedProvenTx,
     tx_inputs: TxInputs,
 ) -> Result<(), VerifyTxError> {
+    info!(
+        tx_id = %candidate_tx.id().inner(),
+        account_id = ?candidate_tx.account_id(),
+        tx_inputs.account_hash = %tx_inputs.account_hash.as_ref().map(ToString::to_string).unwrap_or("None".to_string()),
+        ?tx_inputs.nullifiers,
+        COMPONENT,
+    );
```

Comment:
I'd probably extract transaction ID on line 175 above, but other than that, I'm not sure we need this. If there is an error, we'll get the error message with what didn't match.